### PR TITLE
RUN-5140 IAB.send should return an error if the destination is not valid object

### PR DESCRIPTION
**Pull request by licui3936**
This PR is to validate the identity object passed in IAB.send and all the wrap functions.

### CODE REVIEW

File: src/api/interappbus/interappbus.ts [hunk 1]
```diff
@@ -61,6 +61,9 @@ export default class InterApplicationBus extends Base {
      * @tutorial InterApplicationBus.send
     */
     public send(destination: Identity, topic: string, message: any): Promise<void> {
+        if (typeof destination !== 'object' || typeof destination.uuid !== 'string') {
+            throw new Error('The destination is not a valid identity object');
```
Comment (rdepena) [inline src/api/interappbus/interappbus.ts]: We should reject the promise instead of throwing an exception.

Comment (licui3936) [reply src/api/interappbus/interappbus.ts]: Change it.

**Comment by pbaize** [comment]
Should we just make these async functions instead of returning promise.resolve and promise.reject?

**Comment by licui3936** [comment]
@pbaize agree, i will change it.

File: src/util/validate.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Identity } from '../identity';
+
+export function validateIdentity(identity: Identity): string {
+    let errorMsg;
+
+    if (typeof identity !== 'object' || typeof identity.uuid !== 'string') {
```
Comment (datamadic) [inline src/util/validate.ts]: can we check that if a name is present that it is a string, this eliminates the following class inconsistency: 
```js
fin.Application.wrapSync({name: 123, uuid:'123'}) // totally fine
fin.Application.start({name: 123, uuid:'123', url: location.href}) // this tosses an error
```

Comment (licui3936) [reply src/util/validate.ts]: 'start' function uses app options object, not identity.

**Pull request merged**